lib/unittest.bzl

```diff
@@ -170,9 +180,16 @@ def _make_analysis_test(impl, expect_failure = False, config_settings = {}):
         test_transition = analysis_test_transition(
             settings = changed_settings,
         )
-        attrs["target_under_test"] = attr.label(cfg = test_transition, mandatory = True)
+        attrs["target_under_test"] = attr.label(
```

Comment:
Since this duplication is getting longer, you could probably do something like:

```
    test_transition = None
    if changed_settings:
        test_transition = analysis_test_transition(
            settings = changed_settings,
        )

    attrs["target_under_test"] = attr.label(
        aspects = [_action_retrieving_aspect],
        cfg = test_transition,
        mandatory = True,
    )
```

Not sure which ends up being better, so completely up to you.

Reply:
I prefer the way it currently is, if you have no strong preference. To be re-evaluated if the code gets more complex :)